cmd/nginx-meshctl/main.go

```diff
@@ -2,22 +2,176 @@
 package main
 
 import (
+	"bufio"
+	"io"
+	"log"
 	"os"
+	"path/filepath"
+	"regexp"
+	"runtime"
+	"strings"
 
 	"github.com/nginxinc/nginx-service-mesh/internal/nginx-meshctl/commands"
+
+	"github.com/spf13/cobra"
+	"github.com/spf13/cobra/doc"
+	"golang.org/x/text/cases"
+	"golang.org/x/text/language"
 )
 
 var (
 	pkgName = "nginx-meshctl"
 	version = "0.0.0"
 	commit  = "local"
+	genDocs = ""
 )
 
 func main() {
 	cmd := commands.Setup(pkgName, version, commit)
 	cmd.SilenceUsage = true
 
+	if genDocs != "" {
+		if err := generateMarkdown(cmd); err != nil {
+			log.Fatal(err)
+		}
+	}
+
 	if cmd.Execute() != nil {
 		os.Exit(1)
 	}
 }
+
+func generateCompletion(cmd *cobra.Command, docsPath string) error {
+	cmd.CompletionOptions.HiddenDefaultCmd = false
+	cmd.InitDefaultCompletionCmd()
+	completionCmd, _, err := cmd.Find([]string{"completion"})
+	if err != nil {
+		return err
+	}
+	mdFileFullPath := filepath.Join(docsPath, "nginx-meshctl_completion.md")
+
+	mdFile, err := os.Create(mdFileFullPath)
+	if err != nil {
+		return err
+	}
+	defer mdFile.Close()
+
+	caser := cases.Title(language.English)
+
+	scanner := bufio.NewScanner(strings.NewReader(completionCmd.UsageString()))
+	if _, err = mdFile.WriteString("## " + caser.String(completionCmd.Name()) + "\n"); err != nil {
+		return err
+	}
+
+	for scanner.Scan() {
+		line := strings.ReplaceAll(scanner.Text(), "ke.agarwal", "<user>")
+		if strings.Contains(line, "Usage:") {
+			_, err = mdFile.WriteString("\n```\n")
+		} else if strings.Contains(line, "Available Commands:") {
+			_, err = mdFile.WriteString("```\n\n### Available Commands\n\n```\n")
+		} else if strings.Contains(line, "Global Flags:") {
+			_, err = mdFile.WriteString("```\n\n### Options\n\n```\n")
+		} else if strings.Contains(line, "for more information about") {
+			_, err = mdFile.WriteString("```\n")
+		} else if len(line) > 0 {
+			_, err = mdFile.WriteString(line + "\n")
+		}
+		if err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func generateMarkdown(cmd *cobra.Command) error {
+	cmd.DisableAutoGenTag = true
+
+	// generates a markdown file for each CLI command
+	_, b, _, _ := runtime.Caller(0)
+	docsPath := filepath.Join(filepath.Dir(b), "../../docs/content/reference")
```

Comment:
I would rather have the CLI spit out all of this to stdout, and then the caller of the generator can pipe that output into our docs file. I don't want our CLI itself to write the file.